### ISSUE TITLE
과제 질문

### ISSUE THREAD

**Issue by jkim50104** [ML-2020-spring-KU/ML_2020_spring#2]
이번 Transfer learning을 하다가 든 생각인데요. 데이터의 크고 작음과 데이터의 유사성이 시나리오에 영향을 줍니다. 근데 데이터의 크기의 크고 작음의 기준과 유사성의 기준은 어떻게 봐야 합리적이나요? 특히 두개의 데이터의 유사성을 볼때 둘이 비슷한 이미지를 가져야 유사한건가요? 아니면 데이터 클래스가 비슷해야 유사한 것인가요? 유사성을 판단하는 기준이 어떻게 되나요?

**Comment by hyeonjoong-cho** [comment]
숫자로 표현되는 명확한 기준이 없지 않을까 싶습니다. 

1. 데이터셋의 크기

데이터셋의 크기는 모델의 파라메터 수와 함께 생각해야할 것 같네요. 데이터셋의 크기가 커도 모델의 파라메터수가 훨씬더 크다면 해당 데이터셋으로 모델을 적절히 학습시킬수 없습니다. 따라서, 어떤 모델A가 데이터셋 D로 적절히 pretrain되어 있다면 모델 A를 새로운 데이터셋 D+로 새롭게 학습시킬때 D와 D+의 크기를 비교해볼 수는 있을것 같습니다. 만약 D보다 D+가 적다면 overfitting에 대한 우려를 해볼만 합니다.
통상적으로 사용하는 모델의 파라메터수가 극단적으로 다르지 않아서 통상적으로 사용하는 데이터셋을 살펴보는 것도 좋을것 같네요. CIFAR-100은 100개의 클래스와 클래스별 600의 이미지, ImageNet 데이터셋은 2만개 이상의 클래스, 총14,197,122장의 이미지라고 합니다. 이들을 위해 설계된 모델을 사용할때는 그만큼의 데이터셋이 필요할것 같습니다.

2. 유사성

데이터셋 A와 B가 같은 클래스를 갖고 있다고 가정합시다. (가령 MNIST-1 vs MNIST-2, 혹은 ImageNet-1 vs ImageNet-2. ) 데이터셋 A, B의 클래스가 같다고 하더라도, X들의 분포, Y의 확률분포가 다를 수 있습니다.  분류 문제의 경우 Y의 분포를 비교하는 것은 쉽습니다. Label (Class)별로 히스토그램을 비교하면 되니까요. 반면에 차원이 높은 X의 분포를 비교하는 것은 어려울 수 있습니다. 차원을 줄여서 분포를 비교하는 방법이 필요하겠죠. 분포가 구해지고나면 두 분포간의 거리를 구하는 방법은 다양하게 있습니다. 분포의 거리가 가까운 데이터셋은 유사성이 높다할 수 있고, 분포의 거리가 먼 데이터셋은 유사성이 낮다고 할 수 있습니다.
